.github/workflows/main.yaml

```diff
@@ -0,0 +1,76 @@
+name: Shellcheck, Testing and GitHub Container Registry CI
+on:
+  pull_request:
```

Comment:
Also, how do you plan to determine the docker image tag? It also needs to be updated in the SDL file since it is pulled automatically in console. I was thinking of this workflow:

1. PR: Code changes
2. Update the SDL image tag
3. Merge
4. Build and push the docker image based on the tag in the SDL

Reply:
We want to avoid the double pr for a single change. I suggest to read the SDL file image tag and build the version based on the git diff.